BaseStation/MainWindow/MapTileDownloader.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Net;
+using System.IO;
+using System.Collections.Generic;
+using HuskyRobotics.Utilities;
+
+// make sure there is a folder named MapTiles in the working directory
+// google API key is stored in environment variables
+// https://msdn.microsoft.com/en-us/library/bb259689.aspx#Map map system (its bing but google
+// uses the same system)
+// https://google-developers.appspot.com/maps/documentation/static-maps/intro#Zoomlevels google
+// tile download documentation
+
+namespace HuskyRobotics.UI
+{
+    public static class MapTileDownloadManager
+    {
+        // a class to hold the configuration for the map dowload, formated as specified by google
+        public class Configuration
+        {
+            private Dictionary<String, String> parameters;
+
+            public Configuration(int imgWidth, int imgHeight, int scale, int zoom, Tuple<double, double> coords)
+            {
+                parameters = new Dictionary<String, String>();
+
+                SetLocation(coords);
+                SetZoom(zoom);
+                SetImageSize(imgHeight, imgWidth);
+                SetImageScale(scale);
+                parameters.Add("maptype", "satellite");
+            }
+
+            // adds the image dimensions to the current configuration
+            public void SetImageSize(int width, int height)
```

Comment:
Change these getters/setters to properties.

Reply:
I'm thinking I'll restructure the entire configuration class. I'm not sure if a dictionary is a very good data structure for this because it's harder to extract the different types of data being stored